zerver/views/user_groups.py

```diff
@@ -39,51 +40,34 @@
     user_groups_in_realm_serialized,
 )
 from zerver.lib.users import access_user_by_id, user_ids_to_users
-from zerver.lib.validator import check_bool, check_dict_only, check_int, check_list, check_union
+from zerver.lib.validator import check_bool, check_int, check_list
 from zerver.models import NamedUserGroup, UserGroup, UserProfile
 from zerver.models.users import get_system_bot
 from zerver.views.streams import compose_views
 
 
 def parse_group_setting_value(
-    setting_value: Union[int, Dict[str, List[int]]],
+    setting_value: Union[int, AnonymousSettingGroupDict],
 ) -> Union[int, AnonymousSettingGroupDict]:
     if isinstance(setting_value, int):
         return setting_value
 
-    if len(setting_value["direct_members"]) == 0 and len(setting_value["direct_subgroups"]) == 1:
-        return setting_value["direct_subgroups"][0]
+    if len(setting_value.direct_members) == 0 and len(setting_value.direct_subgroups) == 1:
+        return setting_value.direct_subgroups[0]
 
-    return AnonymousSettingGroupDict(
-        direct_members=setting_value["direct_members"],
-        direct_subgroups=setting_value["direct_subgroups"],
-    )
-
-
-check_group_setting: Validator[Union[int, Dict[str, List[int]]]] = check_union(
-    [
-        check_int,
-        check_dict_only(
-            [
-                ("direct_members", check_list(check_int)),
-                ("direct_subgroups", check_list(check_int)),
-            ]
-        ),
-    ]
-)
+    return setting_value
 
 
 @require_user_group_edit_permission
-@has_request_variables
+@typed_endpoint
 def add_user_group(
     request: HttpRequest,
     user_profile: UserProfile,
-    name: str = REQ(),
-    members: Sequence[int] = REQ(json_validator=check_list(check_int), default=[]),
-    description: str = REQ(),
-    can_mention_group: Optional[Union[Dict[str, List[int]], int]] = REQ(
-        json_validator=check_group_setting, default=None
-    ),
+    *,
+    name: str,
+    members: Json[Sequence[int]],
```

Comment:
Previously we set the default for `members` field as `[]`. I intended to set the same default, but mypy raised error with solution being setting the default to `None`.

Also `members` field is marked as `required` in the API documentation, so might probably be better to just not have any default here.

Reply:
Yeah, no default should be required, so this seems correct.